content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx

```diff
@@ -208,6 +205,10 @@ export class _DiscoveryStreamBase extends React.PureComponent {
     // Get "topstories" Section state for default values
     const topStories = this.props.Sections.find(s => s.id === "topstories");
 
+    if (!topStories) {
+      return null;
+    }
+
     // Extract TopSites to render before the rest and Message to use for header
     const topSites = extractComponent("TopSites");
     const message = extractComponent("Message") || {
```

Comment:
For non-cached, first load we can avoid extractComponent multiple calls if topSites is not null and message & message.type is Message

Reply:
As discussed over slack , optimizing these calls inside selectLayoutRender can be a refactoring in a separate bug